src/reflect/scala/reflect/internal/tpe/TypeComparers.scala

```diff
@@ -180,7 +180,11 @@ trait TypeComparers {
     !settings.isScala211 || methodHigherOrderTypeParamsSameVariance(low, high) || low.variance.isInvariant
 
   def isSameType2(tp1: Type, tp2: Type): Boolean = {
-    def retry(lhs: Type, rhs: Type) = ((lhs ne tp1) || (rhs ne tp2)) && isSameType(lhs, rhs)
+    def retry(lhs: Type, rhs: Type) = ((lhs ne tp1) || (rhs ne tp2)) && {
+      def sameSyms(tp1: Type, tp2: Type) = tp1.typeSymbolDirect eq tp2.typeSymbolDirect
+      val skipRetry = lhs.isInstanceOf[PolyType] && rhs.isInstanceOf[PolyType] && sameSyms(tp1, lhs) && sameSyms(tp2, rhs)
```

Comment:
Could you convince me that we don't need to take into account potential differences in type params / type application due to normalization? I'm not sure how, but I'm also not sure it's impossible that `normalizePlus` could take e.g., `[T, U](T, U)` to `[T, U](U, T)` for one of `rhs`/`lhs`, so that now `[T,U]Swap[U,T] =:= [T,U]Tuple2[T,U]`

Also, this feels easier to follow:
```
      val skipRetry =
        (lhs eq tp1) && (rhs eq tp2) ||
          (lhs.isInstanceOf[PolyType] && rhs.isInstanceOf[PolyType] && sameSyms(tp1, lhs) && sameSyms(tp2, rhs))

      !skipRetry && isSameType(lhs, rhs)
```